### PR TITLE
Adds Amazon Corretto 15 support, changes to `java_certificate`

### PR DESCRIPTION
**Pull request by jakauppila**
### Description

Adds Amazon Corretto 15 support
Adds configurable file cache property to `java_certificate`
Adds `cacerts` property for interacting with java cacerts file for Java 9+

### Issues Resolved

[List any existing issues this PR resolves]

### Check List

- [x] All tests pass. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/TESTING.MD>
- [x] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable

### CODE REVIEW

File: resources/certificate.rb
Comment (jakauppila) [inline resources/certificate.rb:60]: I don't love this property name; any better suggestions?

**Review by jakauppila — COMMENTED** [review]
Is there a reason why the corretto suites didn't fire for dokken?

**Comment by ramereth** [comment]
> Is there a reason why the corretto suites didn't fire for dokken?

Looks like they're commented out for whatever reason: https://github.com/sous-chefs/java/blob/9b5abe4e52f8de571bc5ba97edc00b9d73ac4aa4/.github/workflows/ci.yml#L64-L65

File: resources/certificate.rb
Comment (ramereth) [reply resources/certificate.rb:60]: I can't think of anything better.

**Comment by ramereth** [comment]
@jakauppila looks like you have some issues with a few platforms that you need to work out before merging.

**Review by majormoses — DISMISSED** [review]
LGTM after getting the tests passing for those platforms

**Comment by kitchen-porter** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please include a CHANGELOG entry.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by ramereth** [comment]
@jakauppila please add a change log entry

**Comment by jakauppila** [comment]
So these latest failures just illustrate the problems of hardcoding the values in the cookbook; it looks like AWS has released newer version 8 and 11 behind those "friendly" URLs.

**Review by ramereth — APPROVED** [review]

**Comment by kitchen-porter** [comment]
Released as: [8.6.0](https://github.com/sous-chefs/java/releases/tag/8.6.0)

**Pull request merged**
